File: api/lib/units/tank.rb
require_relative './vehicle.rb'

class Tank < Vehicle
  def initialize square, team
    super
    @type = :tank
    @baseOffense = 2
    @baseDefense = 2
  end

  def assign_threat
    super
    square
      .neighbors(headings: [[0, 1], [1, 1]])
      .each { |square| threaten square }
  end
end

File: api/src/classes/units/Tank.js
const Vehicle = require('./vehicle')

class Tank extends Vehicle {
  constructor(square, team) {
    super(square, team)
    this.type = 'tank'
    this.baseOffense = 2
    this.baseDefense = 2
  }

  assignThreat() {
    super()
    this.square
      .neighbors({headings: [[0, 1], [1, 1]]})
      .forEach(square => threaten(square))
  }
}

module.exports = Tank